internal/config/config.go

```diff
@@ -40,6 +40,7 @@ type ClusterResources struct {
 	BGPAdvs            []metallbv1beta1.BGPAdvertisement `json:"bgpadvertisements"`
 	L2Advs             []metallbv1beta1.L2Advertisement  `json:"l2advertisements"`
 	LegacyAddressPools []metallbv1beta1.AddressPool      `json:"legacyaddresspools"`
+	Communities        []metallbv1beta1.Community        `json:"communities"`
```

Comment:
Nit about the commit message: try to be impersonal.

```
The new logic is tested under config_test.go,
Additional tests were added.
Pools were added to the tests involving BGPAdv, because otherwise it would pass for the wrong reason (such as lack of IPPool definition)
```